parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordWriter.java

```diff
@@ -166,9 +166,16 @@ public long getDataSize() {
   }
 
   private void checkBlockSizeReached() throws IOException {
-    if (recordCount
-        >= recordCountForNextMemCheck) { // checking the memory size is relatively expensive, so let's not do it
-      // for every record.
+    if (recordCount >= rowGroupRecordCountThreshold) {
+      LOG.debug("record count reaches threshold: flushing {} records to disk.", recordCount);
```

Comment:
Maybe better to escalate the log level to INFO? It should not be noisy.

Reply:
For a typical Hadoop YARN cluster, which serves Spark workloads, application logs are preserved for a few days by collecting and aggregating to HDFS. Anyway, it's not a big deal, we can always use parquet-cli to analyze the suspicious Parquet files :)